### DISCUSSION TITLE
Pico 4 experimental release - increasing app / hmd resolution?

### DISCUSSION THREAD

**Discussion by SunSpireUK** [icosa-gallery/open-brush#355] [General]
Hello there! Happy New Year! 

I've recently checked out the experimental Pico 4 release of Open Brush that was showcased by someone on YT. It already runs very well and the added passthrough mode is especially fun! However, the app runs in a very low resolution inside the HMD, the image is somewhat blurry and you get very jagged lines. Clearly this is an early build, so it is understandable.

I could not find much information about the current development process of Open Brush for the Pico 4, or if it was still actively being worked on. I'd hope so! It has the potential of becoming one of the coolest creative apps for the device, period :) And I'm certain that there are other Pico users out there who are looking forward to an improved version, too.

Know any more details you can share? Or anything about what needs to be done to improve the app for Pico 4?

Cheers!
SunSpire

**Comment by mikeskydev** [comment]
Hi and happy new year! Thanks for your interest in the experimental Pico build!

Our Pico builds are currently synchronised with our beta builds so it's fairly close to release. I believe the quality issues you're seeing are sadly due to an issue we experienced while upgrading the Quest version, which had a crashing issue with some custom anti-aliasing we use. We could use supersampling, but our test device (Pico Neo 3) showed that it was struggling more on larger scenes compared to the Quest 2, so I'm hesitant to enable it right now. As Quest is an official release it takes priority right now, but I promise you that when Pico is officially supported we'll make this right! Shouldn't be too long to wait, look for more info towards the end of Jan. Hope this helps and happy to answer any more queries!

**Reply by FictionX** [reply]
Hi

Thank you for your efforts!  I just wanted to comment on this...

AA definitely seems to be missing - but even more importantly, OpenBrush (latest version from SideQuest) is running at 1296x1425 , which I think is the lowest I've seen any program use on the Pico 4.  
(checked using the Pico Metrics Tool)

Even 1440x1584 which seems to be the default on Quest 2 (and many devs just leave as is when they port to Pico 4) results in a very blurry image. The screens are 2160x2160 for each eye - and of course not all apps can run at that high a resolution without it affecting the performance, but looking at other apps, I would think it should be possible to increase it significantly for OpenBrush.

A few examples of what other games/apps use:

Golf 5: 2160x2376  / 60fps
Puzzling Places: 2160x2160 / 72fps
Paradiddle:  2016x2220 / 72fps
V-Speedway: 2016x2217 / 90 fps
Red Matter 2: 1872x2060 / 72fps
Wander: 2592x2851 / 90fps 

Of course none of those are directly comparable to OpenBrush - but it was just to give you an idea of what is possible. The Pico 4 is more powerful than both Quest 2 and Pico 3.

Thanks again for your work!

**Reply by mikeskydev** [reply]
Thanks, I really appreciate this research and it'll help us dial it in for Pico!

**Reply by mikeage** [reply]
Do we have the ability to detect and differentiate between a Pico 3 and a Pico 4?

**Reply by mikeskydev** [reply]
Through Pico SDK yes, which is *bleh*. Not really the OpenXR way either! Though I do wish there was an OpenXR extension for this sort of device reporting.

**Comment by SunSpireUK** [comment]
Hello mikeskydev,

Thank you so much for your in-depth response! I'm looking forward with excitement to all future development of Open Brush, it will be interesting to see what lies ahead. The PCVR (on a Reverb G2) version is awesome, too. It's what I'm using at the moment since I had to return the Pico due to a stuck center pixel on the lcd screen. I'll be happy to do more testing once the replacement unit has arrived :)

**Comment by mikeskydev** [comment]
In #430 we changed the default quality on mobile to be 1x, previously it was 0.9x which would explain the lower resolution here. Yet to do a fine tuning pass on the best settings for each headset, but I can think of a few ways to try it now!

**Comment by SunSpireUK** [comment]
I tried the update you mentioned but did not notice a difference in quality on the Pico 4, except that passthrough no longer works and remains opaque. But I believe that is because you have not officially added that yet, that is, outside of the experimental release I got from elsewhere previously, right?

What happens if the quality was bumped up all the way to 1.5 res? And would it be possible to implement an .ini config setting that let's a user change this according to their preference? I guess the Pico 4 will eventually slow down if the resolution becomes too high or the scene too big, but I'd rather work with a clear, jag-free image than one with fancy fx, such as bloom. Staying excited for more updates :)
